tpm2/test/tpm2_test.go

```diff
@@ -575,6 +575,144 @@ func TestCertify(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	params := Public{
+		Type:       AlgRSA,
+		NameAlg:    AlgSHA256,
+		Attributes: FlagSignerDefault,
+		RSAParameters: &RSAParams{
+			Sign: &SigScheme{
+				Alg:  AlgRSASSA,
+				Hash: AlgSHA256,
+			},
+			KeyBits: 2048,
+		},
+	}
+	signerHandle, signerPub, err := CreatePrimary(rw, HandleOwner, pcrSelection7, emptyPassword, defaultPassword, params)
+	if err != nil {
+		t.Fatalf("CreatePrimary(signer) failed: %s", err)
+	}
+	defer FlushContext(rw, signerHandle)
+
+	subjectHandle, subjectPub, err := CreatePrimary(rw, HandlePlatform, pcrSelection7, emptyPassword, defaultPassword, params)
+	if err != nil {
+		t.Fatalf("CreatePrimary(subject) failed: %s", err)
+	}
+	defer FlushContext(rw, subjectHandle)
+
+	attest, sig, err := CertifyEx(rw, defaultPassword, defaultPassword, subjectHandle, signerHandle, nil, SigScheme{Alg: AlgRSASSA, Hash: AlgSHA256, Count: 0})
+	if err != nil {
+		t.Errorf("Certify failed: %s", err)
+		return
+	}
+
+	attestHash := sha256.Sum256(attest)
+	if err := rsa.VerifyPKCS1v15(signerPub.(*rsa.PublicKey), crypto.SHA256, attestHash[:], sig); err != nil {
+		t.Errorf("Signature verification failed: %v", err)
+	}
+
+	t.Run("DecodeAttestationData", func(t *testing.T) {
+		ad, err := DecodeAttestationData(attest)
+		if err != nil {
+			t.Fatal("DecodeAttestationData:", err)
+		}
+		params := Public{
+			Type:       AlgRSA,
+			NameAlg:    AlgSHA256,
+			Attributes: FlagSignerDefault,
+			RSAParameters: &RSAParams{
+				Sign: &SigScheme{
+					Alg:  AlgRSASSA,
+					Hash: AlgSHA256,
+				},
+				KeyBits: 2048,
+				// Note: we don't include Exponent because CreatePrimary also
+				// returns Public without it.
+				ModulusRaw: subjectPub.(*rsa.PublicKey).N.Bytes(),
+			},
+		}
+		matches, err := ad.AttestedCertifyInfo.Name.MatchesPublic(params)
+		if err != nil {
+			t.Fatalf("AttestedCertifyInfo.Name.MatchesPublic error: %v", err)
+		}
+		if !matches {
+			t.Error("Name in AttestationData doesn't match Public structure of subject")
+		}
+	})
+}
+
+func TestCertifyExAlgNull(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	params := Public{
+		Type:       AlgRSA,
+		NameAlg:    AlgSHA256,
+		Attributes: FlagSignerDefault,
+		RSAParameters: &RSAParams{
+			Sign: &SigScheme{
+				Alg:  AlgRSASSA,
+				Hash: AlgSHA256,
+			},
+			KeyBits: 2048,
+		},
+	}
+	signerHandle, signerPub, err := CreatePrimary(rw, HandleOwner, pcrSelection7, emptyPassword, defaultPassword, params)
+	if err != nil {
+		t.Fatalf("CreatePrimary(signer) failed: %s", err)
+	}
+	defer FlushContext(rw, signerHandle)
+
+	subjectHandle, subjectPub, err := CreatePrimary(rw, HandlePlatform, pcrSelection7, emptyPassword, defaultPassword, params)
+	if err != nil {
+		t.Fatalf("CreatePrimary(subject) failed: %s", err)
+	}
+	defer FlushContext(rw, subjectHandle)
+
+	attest, sig, err := CertifyEx(rw, defaultPassword, defaultPassword, subjectHandle, signerHandle, nil, SigScheme{Alg: AlgNull, Hash: AlgSHA256, Count: 0})
```

Comment:
AlgNull/AlgSHA256 doesn't really make sense as a combination of values to SigScheme - we should rely on default initialization being good enough for the members that don't matter to the use-case and have simply SigScheme{Alg: AlgNull}